File: TestAndroidV2/app/src/main/java/br/com/altran/santander/davidmelo/ui/login/LoginContract.java
package br.com.altran.santander.davidmelo.ui.login;

import br.com.altran.santander.davidmelo.ui.base.MvpPresenter;
import br.com.altran.santander.davidmelo.ui.base.MvpView;

public interface LoginContract {
    interface View extends MvpView {
        void makeLogin();

        void setUp();
    }

    interface Presenter<V extends MvpView> extends MvpPresenter<V> {
        void makeLogin(LoginActivity login, String user, String password);
    }
}

File: BankApp/app/src/main/java/br/com/santander/android/bank/login/presentation/LoginContract.kt
package br.com.santander.android.bank.login.presentation

import br.com.santander.android.bank.core.base.BasePresenter
import br.com.santander.android.bank.core.base.BaseView
import br.com.santander.android.bank.login.domain.model.Account
import br.com.santander.android.bank.login.domain.model.UserAccount

interface LoginContract {

    interface View : BaseView<Presenter> {
        fun showLoading()
        fun hideLoading()
        fun showEmptyUserError()
        fun showEmptyPasswordError()
        fun showInvalidPasswordError()
        fun showOfflineMessage()
        fun showTryAgainMessage()
        fun onLoginSuccess(userAccount: UserAccount)
    }

    interface Presenter : BasePresenter {
        fun requestLogin(login: String, password: String)
        fun saveSession(userAccount: UserAccount)
        fun getSavedSession(): Account?
    }
}